packages/binding-opcua/test/full-opcua-thing-test.ts

```diff
@@ -0,0 +1,624 @@
+                        // maximum: 100,
+                    },
+                    Volume: {
+                        type: "number",
+                        minimum: 0,
+                        maximum: 255,
+                    },
+                },
+                required: ["SongList", "Volume"],
+            },
+            output: {
+                type: "object",
+                properties: {
+                    SoundAndLyrics: {
+                        type: "array",
+                        title: "an array of key value pair containing song as key and lyrics as value",
+                        // minimum: 0,
+                        // maximum: 100,
+                    },
+                },
+                required: ["SoundAndLyrics"],
+            },
+        },
+    },
+};
+
+describe("Full OPCUA Thing Test", () => {
+    let opcuaServer: OPCUAServer;
+    let endpoint: string;
+    before(async () => {
+        opcuaServer = await startServer();
+        endpoint = opcuaServer.getEndpointUrl();
+        console.log("endpoint = ", endpoint);
+    });
+    after(async () => {
+        await opcuaServer.shutdown();
+    });
+
+    it("Z1- should create a servient (produce) with OPCUA client factory", async () => {
+        const servient = new Servient();
+
+        const opcuaClientFactory = new OPCUAClientFactory();
+        servient.addClientFactory(opcuaClientFactory);
+
+        const wot = await servient.start();
+        const thing = await wot.produce(thingDescription);
+
+        thing.expose();
+
+        let temperature = 10;
+        thing.setPropertyReadHandler("temperature", async () => temperature);
+
+        const expThing = thing as ExposedThing;
+        const readHandler = expThing.propertyHandlers.get("temperature").readHandler;
+        expect(readHandler, "must have a readHandler");
+        const temperatureCheck1 = await readHandler();
+        expect(temperatureCheck1).to.equal(10);
+
+        temperature = 100;
+
+        const temperatureCheck2 = await readHandler();
+        expect(temperatureCheck2).to.equal(100);
+
+        await servient.shutdown();
+    });
+
+    async function makeThing() {
+        const servient = new Servient();
+
+        const opcuaClientFactory = new OPCUAClientFactory();
+
+        servient.addClientFactory(opcuaClientFactory);
+
+        const wot = await servient.start();
+
+        const thing: WoT.ConsumedThing = await wot.consume(thingDescription);
+
+        console.debug(thing.getThingDescription().properties);
+
+        return { thing, servient };
+    }
+    async function doTest(thing: WoT.ConsumedThing, propertyName: string, localOptions: InteractionOptions) {
+        console.log("------------------------------------------------------");
+        try {
+            const content = await thing.readProperty(propertyName, localOptions);
+            const dataSchemaValue = await content.value();
+            const json = dataSchemaValue.valueOf();
```

Comment:
I was reviewing again the PR and this detail got me worried. Why are you calling `valueOf`?? the default behavior should be that the value function returns an application value not opc-ua payload information. @danielpeintner what do you think?

Reply:
I tend to agree. I mean for primitive values like strings it does not matter but for an object it makes a a difference. 
Is there a reason for calling `valueOf()` we are not aware of? (for the test?)

Reply:
valueOf() doesn't seem to be needed at all, I've cleaned this.